charts/kafka-operator/templates/operator-deployment.yaml

```diff
@@ -34,8 +34,8 @@ spec:
         app: prometheus
         component: alertmanager
     spec:
-      {{- if .Values.rbac.enabled }}
-      serviceAccountName: {{ include "kafka-operator.fullname" . }}-operator
+      {{- if .Values.serviceAccount.name }}
+      serviceAccountName: {{ .Values.serviceAccount.name }}
       {{- end }}
```

Comment:
Please move this part under `_helpers.tpl` and solve the logic there, will result in much cleaner code.
See as an example the Grafana helm chart:
https://github.com/helm/charts/blob/0219c128a8c1be5ea39842e40e69295664001efc/stable/grafana/templates/_helpers.tpl#L34